scripts/stackstorm.js

```diff
@@ -248,7 +250,12 @@ module.exports = function(robot) {
         robot.logger.info(command_factory.st2_hubot_commands.length + ' commands are loaded');
       })
       .catch(function (err) {
-        robot.logger.error('Failed to retrieve commands from "%s":\n%s', env.ST2_API_URL, err);
+        var error_msg = 'Failed to retrieve commands from "%s": %s';
+        robot.logger.error(util.format(error_msg, env.ST2_API_URL, err.message));
+        // Exit in the first time when loadCommands is called with invalid ST2_API_URL, ST2_API_KEY or ST2_AUTH_TOKEN.
+        if (opts.exitOnFailure) {
+          logErrorAndExit(err);
```

Comment:
@jinpingh This change just brings back the logic that we agreed is not a solution as it will stop on _any_ error that potentially could be recovered. @blag just reverted it in previous commits.

For example, what if server returned temporary unavailable error or networking went down? Stopping hubot on all of those cases is not a solution. In this specific case, we need to stop only on Unauthorized. Trying to catch granular errors is a better solution in this situation.

Please see again:
https://stackstorm.slack.com/archives/G0HP0ML0M/p1559872202069500?thread_ts=1559839097.055600&cid=G0HP0ML0M
https://stackstorm.slack.com/archives/G0HP0ML0M/p1559873365070300?thread_ts=1559839097.055600&cid=G0HP0ML0M

Please ask any questions if you need to.

Reply:
Alright, here is the log:
```
ERROR Failed to retrieve commands from "http://donating-angelfish-st2api:9101/": Unauthorized
ERROR Stream error: Event { type: 'error', status: 401 }
```
Can we stop hubot only if we see `401` or `Unauthorized`?

Reply:
This is only called once during`st2chatops` initialization. Did test after `st2chatops` is up running, then issue `service st2stream stop`,  after few minutes  issue `service st2stream start`, and `st2chatops` is back up on line without  issue.

Reply:
I can check if it is ` status: 401` issue then exit, but we will miss to catch invalid `ST2_API_URL`

Reply:
Talking about unresolved API urls, - to avoid breaking anything else and going back and forward in this PR I'm OK to keep that for now in it's previous state because DNS lookup could be a temporary failure. It's not a blocking loop issue and not unrecoverable error like it was in https://github.com/StackStorm/st2chatops/issues/124 case.

Reply:
@jinpingh If we exit on all errors only on first run, `ST2_AUTH_TOKEN` could expire, while `ST2_API_KEY` could be revoked or changed meaning we won't be able to catch Unauthorized errors on next re-loads and st2chatops keeps running even when i got Unathorized response.
Unauthorized issues are important to fix as many users had problems with that before.

Please revert to previous commit and add requested `401/Unauthorized` granular check as well as double-verify/improve the tests cases if needed, because we'll close https://github.com/StackStorm/st2tests/pull/170 by incorporating tests in this PR.

Let's please do it and finish this PR.